include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -597,6 +687,19 @@ template <typename T>
 inline ConstHardwareDevice EpDeviceImpl<T>::Device() const {
   return ConstHardwareDevice(GetApi().EpDevice_Device(this->p_));
 }
+
+template <typename T>
+inline ConstMemoryInfo EpDeviceImpl<T>::GetMemoryInfo(OrtDeviceMemoryType memory_type) const {
+  const auto* mem_info = GetApi().EpDevice_MemoryInfo(this->p_, memory_type);
+  return ConstMemoryInfo{mem_info};
+}
+
+template <typename T>
+inline SyncStream EpDeviceImpl<T>::CreateSyncStream(ConstKeyValuePairs stream_options) const {
```

Comment:
I see that many functions take in the C types as parameters instead of the C++ type. Should this be `const OrtKeyValuePairs*` for consistency and flexibility?

Reply:
I vote C++ type.